File: praca-inzynierska-frontend/src/app/app/model/dto/ShiftDto.ts
import { ShiftType } from "../ShiftType";
import { TimePeriodDto } from './TimePeriodDto';
import { TimeDto } from './TimeDto';
import { Utils } from '../../shared/utils/utils';

export class ShiftDto {

    constructor(
        public id: number = 0,
        public employeeId: number,
        public positionId: number,
        public start: Date,
        public finish: Date,
        public creationDate: Date,
        public shiftType: ShiftType
    ) {}

    get period(): TimePeriodDto {
        return new TimePeriodDto(
            TimeDto.fromDate(this.start),
            TimeDto.fromDate(this.finish)
        )
    }

    static copy(of: ShiftDto): ShiftDto {
        return new ShiftDto(
            of.id,
            of.employeeId,
            of.positionId,
            new Date(of.start),
            new Date(of.finish),
            new Date(of.creationDate),
            of.shiftType
        )
    }

    addTimeZoneToDates(): ShiftDto {
        return new ShiftDto(
            this.id, this.employeeId, this.positionId,
            Utils.fixDateToBackendFormat(this.start),
            Utils.fixDateToBackendFormat(this.finish),
            this.creationDate, this.shiftType
        )
    }

    equalPeriods(other: ShiftDto) {
        return this.period.compare(other.period) == 0
    }
}

File: praca-inzynierska-backend/application/src/main/kotlin/jwer/backend/model/dto/ShiftDto.kt
package jwer.backend.model.dto

import jwer.backend.model.entity.Shift
import jwer.backend.model.entity.ShiftType
import java.time.LocalDateTime

data class ShiftDto (
        val id: Long = 0,
        val employeeId: Long,
        val positionId: Long,
        var start: LocalDateTime,
        var finish: LocalDateTime,
        val creationDate: LocalDateTime?,
        val shiftType: ShiftType?
) {
    constructor(s: Shift)
    : this(s.id, s.employee.id, s.position.id, s.period.start, s.period.finish, s.creationDate, s.shiftType)

    fun boundOverlap(other: ShiftDto): Boolean {
        return start == other.finish || finish == other.start
    }
}